### DISCUSSION TITLE
Bake機能の追加について

### DISCUSSION THREAD

**Discussion by ShimpeiKojima** [hrntsm/HoaryFox#85] [Ideas]
こんにちは
Bake機能の追加について要望があります。

断面符号毎にレイヤーを分けてBakeしたいと考えております。
現在はタグ表示の機能のみなので（機能追加されていたらすみません）
部材種別ごとのBakeしかできない状況かと思います。

ワンボタンで符号毎（断面ごと）のBakeなどできないでしょうか？

**Comment by hrntsm** [comment]
これは例えば [col Name tag](https://hiron.dev/HoaryFox/docs/Component/NameTag#column-name-tag) コンポーネントを例にとると、

- コンポーネントの入力に bool がつく
  - これをtrue にすると text2d で Rhino 側に Bake される  
- 符号は C１ なら C1 というレイヤーに C2 なら C2 レイヤーにBake される

といった機能のイメージでしょうか？

**Reply by ShimpeiKojima** [reply]
すみません、僕の説明不足でした。

レイヤー毎にBakeをしたいのはBrepやLineで、特に符号のテキストをBakeしたいわけではないです。

例えばTag情報がテキストデータで部材毎に出力されていれば、既存コンポーネントでパスを分けて管理できるかと思い、タグ表示の機能のみという文を書きました。

hironさんの箇条書き２点目の考えかたで、Brep、LineがBakeされるイメージです。

**Reply by hrntsm** [reply]
画像のような形で符号ごとに Brep なり Line が Bake されるイメージでしょうか？
![image](https://user-images.githubusercontent.com/23289252/111623230-70a99580-882d-11eb-9388-0c83bc9a6db5.png)

**Reply by ShimpeiKojima** [reply]
その通りです！

**Reply by hrntsm** [reply]
実装検討します！
ありがとうございます。

**Comment by hrntsm** [comment]
試しに実装してみました。
まだテストビルドですがこんな感じでどうでしょうか？
https://github.com/hrntsm/HoaryFox/suites/2304546444/artifacts/48348030

Geomtry に BrepBaker を実装しています。（アイコンはまだです。）
![image](https://user-images.githubusercontent.com/23289252/111872784-41338e00-89d1-11eb-8bab-fefd6f7ab00e.png)

動作状態は、以下の一連のツイートを参照してください。
https://twitter.com/hiron_rgkr/status/1373224270596169728?s=20

**Reply by ShimpeiKojima** [reply]
Twitterで拝見しました！

希望通りの機能です！

お忙しいのにご対応いただきありがとうございます！

**Reply by hrntsm** [reply]
確認ありがとうございます。
正式リリース版の Food4Rhino へアップはもう少し整理したものを上げるのでとりあえずこちらでお願いします。

**Comment by hrntsm** [comment]
こちらの変更を Stb2Line にも適応して HoaryFox ver1.3.1 としてリリースしました。
https://github.com/hrntsm/HoaryFox/releases/tag/v1.3.1
